### PR TITLE
Add pod-name hubble metrics context for pod name label without namespace

### LINKED ISSUE #23122: Hubble metric contexts discrepancy
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

Looking at https://github.com/cilium/cilium/commit/0139215e761bdf698366b79b5a59dee97a7f9841, I see that there's now `app` and `workload-name` contexts, which is much better than `pod-short`.

However, `pod` and `pod-short` include the namespace in the format `<ns>/<pod>` but `app` and `workload-name` don't include the namespace.

There should be an option to get `pod` label context without the namespace prefix.

### Cilium Version

1.13

### Kernel Version

N/A

### Kubernetes Version

N/A

### Sysdump

_No response_

### Relevant log output

_No response_

### Anything else?

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by chancez**
This fixes an inconsistency in the source/destination metrics labels, where currently, `pod` and `pod-short` include the namespace in the label, where the `app` and `workload-name` do not. For consistency, this PR adds a `pod-name` option which doesn't include the namespace. 

Fixes: #23122 

I've marked this for 1.13 because `app` and `workload-name` were introduced for 1.13 and I believe it's important we have consistency for these metrics, otherwise dashboards and alerts may be inconsistent depending on the options your using.

```release-note
Add pod-name hubble metrics context for pod name label without namespace
```

### CODE REVIEW

File: Documentation/observability/metrics.rst
Comment (ghost) [inline Documentation/observability/metrics.rst]: Can it then recommend `workload-name|pod-name`?

Comment (raphink) [reply Documentation/observability/metrics.rst]: Or even `app|workload-name|pod-name`? That seems like the closest to what Hubble does, no?

Comment (chancez) [reply Documentation/observability/metrics.rst]: Good idea.

Comment (ghost) [reply Documentation/observability/metrics.rst]: > Or even app|workload-name|pod-name?

I'm not sure about that, it relies on users setting meaningful labels consistently, which is a bold assumption. workload-name and pod-name are given by Kubernetes.

**Review by ghost — APPROVED** [review]
Looks good, thanks!

**Review by raphink — APPROVED** [review]
LGTM

**Review by qmonnet — APPROVED** [review]
Thanks!

**Comment by qmonnet** [comment]
This is unlikely to make it to v1.13.0 though, is this an issue to add it in a patch release?

File: pkg/hubble/metrics/api/context.go
Comment (kaworu) [inline pkg/hubble/metrics/api/context.go]: Should we copy the documentation about `ContextPod` here and add a note about the namespace prefix in `ContextPod`?

**Review by kaworu — APPROVED** [review]
Thanks @chancez, one nit comment on the way but LGTM.

**Comment by raphink** [comment]
@qmonnet I think it'd be good to get it added early in the 1.13 lifecycle, because at the moment `app|workload-name|pod` has an inconsistent behavior.

**Comment by qmonnet** [comment]
> @qmonnet I think it'd be good to get it added early in the 1.13 lifecycle, because at the moment `app|workload-name|pod` has an inconsistent behavior.

@raphink Makes sense, but in that case I'd recommend reaching out to the TopHat/maintainers in that case, because my understanding was that only release blockers were now backported to 1.13 before the oncoming release? /Cc @ldelossa

File: pkg/hubble/metrics/api/context.go
Comment (chancez) [reply pkg/hubble/metrics/api/context.go]: Ah yeah, I forgot to finish writing the doc comment.

**Review by joestringer — APPROVED** [review]
It's a bit odd that the `Documentation/observability/metrics.rst` requires sig-policy signoff but the changes there LGTM. Maybe we can look at splitting that doc up so that we don't pull in unrelated codeowners for such review.

I see the `upgrade-impact` label but it's not obvious to me how this impacts upgrade, other than that any tools that consume Hubble flows will see some of the fields with slightly different content.

This seems useful enough for consistency and low enough risk that a backport to 1.13 this early in the patch cycle is OK to me.

**Comment by qmonnet** [comment]
I was probably a bit quick to add the `upgrade-impact` label, I'll remove it.

**Comment by joestringer** [comment]
runtime tests failed, @chancez can you triage + file issues if necessary + re-run the tests?

**Comment by chancez** [comment]
@joestringer Sure, it timed out in the `github.com/cilium/cilium/pkg/datapath/linux` tests.

Seems like it timed out in `/home/vagrant/go/src/github.com/cilium/cilium/pkg/datapath/linux/node_linux_test.go:2131 +0x19fc`, but beyond that I'm not sure how much else I can debug. I'll file an issue with the details about the timeout.

**Comment by joestringer** [comment]
Reviews are in, tests passed, LGTM thanks!

**Pull request merged**
